Mage.Sets/src/mage/cards/n/NeeraWildMage.java

```diff
@@ -0,0 +1,115 @@
+package mage.cards.n;
+
+import mage.ApprovingObject;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SpellCastControllerTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.stack.Spell;
+import mage.players.Player;
+
+import java.util.UUID;
+
+/**
+ * @author rjayz
+ */
+public final class NeeraWildMage extends CardImpl {
+
+    public NeeraWildMage(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{U}{R}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.ELF);
+        this.subtype.add(SubType.SHAMAN);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(7);
+
+        // Whenever you cast a spell, you may put it on the bottom of its owner's library.
+        // If you do, reveal cards from the top of your library until you reveal a nonland card.
+        // You may cast that card without paying its mana cost.
+        // Then put the rest on the bottom of your library in a random order.
+        // This ability triggers only once each turn.
+        this.addAbility(new SpellCastControllerTriggeredAbility(new NeeraWildMageEffect(), StaticFilters.FILTER_SPELL_A,
+                true, true).setTriggersOnce(true));
+    }
+
+    private NeeraWildMage(final NeeraWildMage card) {
+        super(card);
+    }
+
+    @Override
+    public NeeraWildMage copy() {
+        return new NeeraWildMage(this);
+    }
+}
+
+class NeeraWildMageEffect extends OneShotEffect {
+
+    public NeeraWildMageEffect() {
+        super(Outcome.Neutral);
+        staticText = "you may put it on the bottom of its owner's library. If you do, reveal cards from the top of your library until you reveal a nonland card. You may cast that card without paying its mana cost. Then put the rest on the bottom of your library in a random order. This ability triggers only once each turn.";
+    }
+
+    public NeeraWildMageEffect(final NeeraWildMageEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Spell spell = game.getStack().getSpell(targetPointer.getFirst(game, source));
+        if (spell == null) {
+            return false;
+        }
+
+        Player spellController = game.getPlayer(spell.getControllerId());
+        if (spellController == null) {
+            return false;
+        }
+
+        spellController.putCardsOnBottomOfLibrary(spell, game, source, true);
```

Comment:
you should check that the card was actually put on the bottom of the library. This will return a bool. You should immediately return false if that bool is false and

Reply:
Should a similar check be done for putting the revealed cards on the bottom of the library at the end of this method?

Reply:
No, that doesn't make a difference for the effect. As I understand it, you return true if any portion of the effect has been successfully applied